crates/vfs-notify/src/lib.rs

```diff
@@ -103,21 +103,39 @@ impl NotifyActor {
                         let config_version = config.version;
 
                         let n_total = config.load.len();
-                        self.send(loader::Message::Progress { n_total, n_done: 0, config_version });
+                        self.send(loader::Message::Progress {
+                            n_total,
+                            n_done: 0,
+                            config_version,
+                            file: None,
+                        });
 
                         self.watched_entries.clear();
 
                         for (i, entry) in config.load.into_iter().enumerate() {
+                            self.send(loader::Message::Progress {
+                                n_total,
+                                n_done: i,
+                                config_version,
+                                file: None,
+                            });
                             let watch = config.watch.contains(&i);
                             if watch {
                                 self.watched_entries.push(entry.clone());
                             }
-                            let files = self.load_entry(entry, watch);
+                            let files =
+                                self.load_entry(entry, watch, |file| loader::Message::Progress {
+                                    n_total,
+                                    n_done: i,
+                                    file: Some(file),
+                                    config_version,
+                                });
                             self.send(loader::Message::Loaded { files });
                             self.send(loader::Message::Progress {
                                 n_total,
                                 n_done: i + 1,
                                 config_version,
+                                file: None,
```

Comment:
With this we now duplicate all loaded messages which seems unnecessary. Instead of having `load_entry` take a callback that triggers message sending we can just set the file field here no? That is, set it to the dir thats being loaded, and nothing if its a file set. From what I can tell, we don't set entries to files right now anyways (and probably won't use it much in the future either).

I worry especially about sending an update for every single file, that is very imperformant and we have already gotten complains before about r-a being too spammy with update notifications which this would make even worse.

Reply:
This only sends updates for top-level directories when the `Directories` variant is used, not every single file. It does send notifications for each file in the `Files` variant, but given those are listed explicitly as a `Vec` I expect them to not be enormous. Am I missing some context here?

Reply:
It'd probably be fine to remove the notifications for the `Files` variant if you'd like, though I don't love the inconsistency.

Reply:
Sorry for the long wait, I'm still of the opinion that this is too spammy. There are clients (editors) out there that just can't deal with a bunch of messages, and as you've stated inthe PR description, 

> This makes it a lot easier to tell that `rust-analyzer` isn't crashed, it's just trying to scan a huge directory.

So I think only showing it for dirs makes the most sense

Reply:
OK, changed it to only send notifications for directories.

Reply:
Your force push didn't seem to change anything